lambda/handler.go

```diff
@@ -99,20 +99,33 @@ func WithEnableSIGTERM(callbacks ...func()) Option {
 	})
 }
 
-func validateArguments(handler reflect.Type) (bool, error) {
-	handlerTakesContext := false
-	if handler.NumIn() > 2 {
-		return false, fmt.Errorf("handlers may not take more than two arguments, but handler takes %d", handler.NumIn())
-	} else if handler.NumIn() > 0 {
+// handlerTakesContext returns whether the handler takes a context.Context as its first argument.
+func handlerTakesContext(handler reflect.Type) (bool, error) {
+	switch handler.NumIn() {
+	case 0:
+		return false, nil
+	case 1:
 		contextType := reflect.TypeOf((*context.Context)(nil)).Elem()
 		argumentType := handler.In(0)
-		handlerTakesContext = argumentType.Implements(contextType)
-		if handler.NumIn() > 1 && !handlerTakesContext {
+		if argumentType.Kind() != reflect.Interface {
+			return false, nil
+		}
+		if argumentType.NumMethod() == 0 {
```

Comment:
This check doesn't appear in the `case 2:` branch, can it be removed?

Reply:
I'm guessing this check is to keep handlers like `func(event any) {}` valid. If that's the case, it's worth adding a comment!

Reply:
yes.
If the first argument is an interface, the handler may be `func(event any)` or `func(ctx context.Context)`.
We should accept both.